src/Shells/StrixMusic.Shells.ZuneDesktop/Controls/Views/Collection/ZuneTrackCollection.cs

```diff
@@ -16,154 +20,182 @@ namespace StrixMusic.Shells.ZuneDesktop.Controls.Views.Collection
     /// <summary>
     /// Zune implmenation for the <see cref="ZuneTrackCollection"/>.
     /// </summary>
-    public class ZuneTrackCollection : TrackCollection
+    [INotifyPropertyChanged]
+    public partial class ZuneTrackCollection : CollectionControl<ZuneTrackCollectionItem, ZuneTrackItem>
     {
+        private ObservableCollection<ZuneTrackCollectionItem> _trackItems = new();
+
+        public ZuneTrackCollection()
+        {
+            TrackItems = new ReadOnlyObservableCollection<ZuneTrackCollectionItem>(_trackItems);
+
+            Loaded += ZuneTrackCollection_Loaded;
+            Unloaded += ZuneTrackCollection_Unloaded;
+        }
+
+        private void AttachEvents(ZuneTrackCollectionItem item)
+        {
+            item.PropertyChanged += Item_PropertyChanged;
+        }
+
+        private void DetachEvents(ZuneTrackCollectionItem item)
+        {
+            item.PropertyChanged -= Item_PropertyChanged;
+        }
+
         /// <summary>
-        /// Holds the instance of a artist column textblock.
+        /// The backing dependency property for <see cref="Collection" />.
         /// </summary>
-        public TextBlock? PART_ArtistColumn { get; private set; }
+        public static readonly DependencyProperty CollectionProperty =
+            DependencyProperty.Register(nameof(Collection), typeof(ITrackCollectionViewModel), typeof(ZuneTrackCollection), new PropertyMetadata(null, (d, e) => ((ZuneTrackCollection)d).OnCollectionChanged((ITrackCollectionViewModel?)e.OldValue, (ITrackCollectionViewModel?)e.NewValue)));
 
         /// <summary>
-        /// Holds the <see cref="ZuneTrackCollection"/> listview.
+        /// The collection to display.
         /// </summary>
-        public ListView? PART_Selector { get; private set; }
+        public ITrackCollectionViewModel? Collection
+        {
+            get => (ITrackCollectionViewModel?)GetValue(CollectionProperty);
+            set => SetValue(CollectionProperty, value);
+        }
 
         /// <summary>
-        /// Creates a new instace for <see cref="ZuneTrackCollection"/>.
+        /// The track items being displayed in the UI. Contains additional functionality specific to this view.
         /// </summary>
-        public ZuneTrackCollection()
+        public ReadOnlyObservableCollection<ZuneTrackCollectionItem> TrackItems { get; }
+
+        /// <summary>
+        /// If true, there is only 1 distinct track artist for all tracks in the provided <see cref="Collection"/>.
+        /// </summary>
+        public bool AllTrackArtistsAreTheSame => TrackItems.Any(x => x.ShouldShowArtistList);
+
+        /// <inheritdoc/>
+        protected override async Task LoadMore()
         {
+            if (Collection is null)
+                return;
+
+            if (!Collection.PopulateMoreTracksCommand.IsRunning)
+                await Collection.PopulateMoreTracksCommand.ExecuteAsync(25);
```

Comment:
Although not related to this PR, but this seems hacky and might need a refactor in future.

Reply:
Copied from the existing `TrackCollection`, it shouldn't have issues but they can be fixed if there are.